pkg/tfgen/generate_schema.go

```diff
@@ -252,7 +255,38 @@ func genPulumiSchema(pack *pkg, name tokens.Package, version string,
 		info:           info,
 		renamesBuilder: renamesBuilder,
 	}
-	return g.genPackageSpec(pack)
+	pulumiPackageSpec, err := g.genPackageSpec(pack)
+	if err != nil {
+		return pschema.PackageSpec{}, err
+	}
+
+	if info.MuxWith != nil {
+		if info.MetadataInfo == nil {
+			return pschema.PackageSpec{},
+				fmt.Errorf("ProviderInfo.MetadataInfo is required and cannot be nil for muxed providers")
+		}
+		muxSchemas := make([]pschema.PackageSpec, len(info.MuxWith)+1)
+		muxSchemas[0] = pulumiPackageSpec
+		for i, v := range info.MuxWith {
+			spec, err := v.GetSpec()
+			if err != nil {
+				return pschema.PackageSpec{}, err
+			}
+			// TODO: do we have to enforce name equality between the main
+			// provider and a mixin (spec.Name == pulumiPackageSpec.Name)?
```

Comment:
@iwahbe I wasn't sure if we really want to enforce this. Actually it makes totally sense, on the other hand: why not mix and match different providers into one single provider.

Reply:
Thinking about it a bit more, I think that this will already be enforced by schema validation. `pulumiPackageSpec.Name` is not enforced (and doesn't need to be), but if the package component of tokens don't match up, schema validation will fail.

Reply:
Done